scripts/lib/CIME/case/case.py

```diff
@@ -980,10 +980,11 @@ def configure(self, compset_name, grid_name, machine_name=None,
 
         for nodename in nodenames:
             value = machobj.get_value(nodename, resolved=False)
-            type_str = self.get_type_info(nodename)
-            if type_str is not None:
-                logger.debug("machine nodname {} value {}".format(nodename, value))
-                self.set_value(nodename, convert_to_type(value, type_str, nodename))
+            if value:
```

Comment:
Bug fix here - when --machine is specified on command line not having a nodename_regex field should not be an error.